### DISCUSSION TITLE
Filtering the Items Array

### DISCUSSION THREAD

**Discussion by EmoCthulhu** [EddieDover/fvtt-party-sheet#9] [Ideas]
Ok, you're a genius and I'm a retard. The only way I tested accessing things under system in the items array was to do `items => system` which resulted in:

![ItemsArray002](https://github.com/EddieDover/fvtt-party-sheet/assets/148644976/1155ae34-c937-4c72-9664-41e465beee7b)

Which is as dumb as doing something as direct and having the text be '"Stuff: system"`.

So anyway, your object-loop already does what's needed to get things out of the system area:

```json
            {
                "name": "Talents/Attacks",
                "type": "object-loop",
                "align": "left",
                "valign": "top",
                "maxwidth": 200,
                "header": "show",
                "text": "items => name - Damage: {b} system.damage {/b}{nl}"
            }
```
Does this:

![ItemsArray001](https://github.com/EddieDover/fvtt-party-sheet/assets/148644976/d67725d9-1378-4326-9f1f-06591e41ca98)

Now the issue is, we need to filter by type. And who knows, someone might end up requesting filtering by name, I don't know. Because the way things are in the Star Trek ships, Phasers and Torpedoes are type "starshipweapon", Tractor Beams are type "item", and Talents are type "talent". Only "starshipweapon" types actually have system.damage.

**Comment by EmoCthulhu** [comment]
So, I'm thinking that it could be as simple as a optional field under "type":

```json
{
    "name": "Attacks",
    "type": "object-loop",
    "filter": "type",
    "header": "show",
    "text": "items{starshipweapon} => name - Damage: {b} system.damage {/b}{nl}"
}
```

That's a great world. But it restricts everything to one column each. So in my oh, my, god you are the man world... Would it be at all possible to run multiple object loops in one "text" field? 

```json
{
    "name": "Attacks/Talents",
    "type": "object-loop",
    "filter": "type",
    "header": "show",
    "text": "{u}Attacks:{/u}{nl} items{starshipweapon} => name - Damage: {b} system.damage {/b}{nl} || {u}Extra Cargo:{/u}{nl} items{item} => name {nl} || {u}Talents:{/u}{nl} items{talent} => name {nl}"
}
```

I just imagined double pipes as separators, but whatever works...

**Comment by EmoCthulhu** [comment]
Oh, one last thing, and I'm not too concerned about this, but I've had one request (from the guy doing Fallout and Star Trek) that some of these arrays display as a dropdown. So thinking about fallout:

```json
{
    "name": "Perks",
    "type": "object-loop",
    "filter": "perk",
    "header": "show",
    "text": "{dropdown} items{perk} => name {b} system.level {/b}{nl}"
}
```

**Reply by EddieDover** [reply]
Dropdown like a select box or dropdown like an accordion?

**Comment by EmoCthulhu** [comment]
Accordion style might be perfect, but I think that would expand the row while open wouldn't it? Select box wouldn't be as slick, but the dropdown wouldn't need to resize the table. Based on his request, I imagined something like {dropdown:Perks} where Perks would be the title and first entry and show when it's closed, clicking that or the + (if accordion) or down arrow (if selectable), would then show all the character's perks. They wouldn't be selectable to end up with one of the perks showing, just a way to see all the perks without having 10-15 line rows for every character. Accordion is slickest, so if that's possible that would be great... but if makes you say "do you realize what you are asking for?" do select box...
